packages/primitives/radio/src/Root.tsx

```diff
@@ -30,27 +39,24 @@ export const RadioInput = component(
   accessibilityLabel,
   value,
   isDisabled,
-  onChange,
+  isVisible,
 }) => (
-  <input
-    type="radio"
-    id={id}
-    name={groupName}
-    key={key || groupName + id}
-    checked={groupValue === value}
-    value={value}
-    aria-labelledby={accessibilityLabelBy.length > 0 ? accessibilityLabelBy.join(' ') : undefined}
-    aria-label={accessibilityLabel}
-    disabled={isDisabled}
-    style={styles}
-    onChange={(evt) => {
-      setGroupValue(evt.currentTarget.value)
-
-      if (typeof onChange === 'function') {
-        onChange(evt)
-      }
-    }}
-  />
+  <label>
+    {children}
+    <input
+      type="radio"
+      id={id}
+      name={groupName}
+      key={key || groupName + id}
+      checked={groupValue === value}
+      value={value}
+      aria-labelledby={accessibilityLabelBy.length > 0 ? accessibilityLabelBy.join(' ') : undefined}
```

Comment:
Generating same result data between renders in animation. This type of data more likely be static for the whole life of the instance.